app/client-v2/src/utils/projectStatus.ts

```diff
@@ -1,7 +1,50 @@
-import type { StatusTypes } from "@/types/projectTypes";
+import { COMPLETE_STATUS_TYPES, type AnalysisStatus, type StatusTypes } from "@/types/projectTypes";
 
 export const hasSampleFailed = (statusType: StatusTypes | undefined, cluster: string | undefined) =>
   statusType === "failed" || (statusType === "finished" && !cluster);
 
 export const hasSamplePassed = (statusType: StatusTypes | undefined, cluster: string | undefined) =>
   statusType === "finished" && !!cluster;
+
+export const hasMicroreactClusterPassed = (
+  microreactClusterStatuses: Record<string, StatusTypes> | undefined,
+  cluster: string | undefined
+): boolean => {
+  if (microreactClusterStatuses && cluster) {
+    return microreactClusterStatuses[cluster] === "finished";
+  }
+  return false;
+};
+
+export const hasMicroreactClusterFailed = (
+  status: AnalysisStatus | undefined,
+  cluster: string | undefined
+): boolean => {
+  // occurs when another sample has passed but this one fails
```

Comment:
Do you mean when it's failed to assign a cluster, hence it being falsy?
```suggestion
  // occurs when this sample could not be assigned to a cluster but at least one other sample was assigned
```

Should we show the microreact job as "failed" in that case? It's more like we couldn't run it because a prerequisite failed. How about "cancelled"?

Reply:
yup cancelled does make sense here.. ive made ticket as this will need a bit more changes https://mrc-ide.myjetbrains.com/youtrack/issue/bacpop-204/When-assign-fails-change-network-microreact-to-cancelled-not-failed